packages/stream/solana/types.ts

```diff
@@ -326,18 +326,17 @@ export class Contract implements Stream {
     this.withdrawalFrequency = stream.withdrawFrequency.toNumber();
   }
 
-  unlocked(currentTimestamp: number, decimals: number): number {
-    const deposited = getNumberFromBN(this.depositedAmount, decimals);
+  unlocked(currentTimestamp: number): BN {
+    const deposited = this.depositedAmount;
 
-    if (currentTimestamp < this.cliff) return 0;
+    if (currentTimestamp < this.cliff) return new BN(0);
```

Comment:
This is agnostic for chain, should be separated

Reply:
Yeah, this class needs a rework, we are currently on dev beta release line, I'll fix it with proper release.